provider/consumer.py

```diff
@@ -84,7 +91,7 @@ def run(self):
                     logging.debug("Consumed message: {}".format(str(message)))
                     messageSize += len(message.value)
                     fieldsToSend = {
-                        'value': message.value,
+                        'value': self.parseMessage(message.value),
```

Comment:
The function returns the right thing depending on whether the message should and can be parsed, and this is nice and easy to read. Perhaps instead I'll change the name to `parseMessageIfNeeded`